internal/providers/openstack/openstack.go

```diff
@@ -44,11 +44,16 @@ const (
 )
 
 var (
-	metadataServiceUrl = url.URL{
+	ipv4MetadataServiceUrl = url.URL{
 		Scheme: "http",
 		Host:   "169.254.169.254",
 		Path:   "openstack/latest/user_data",
 	}
+	ipv6MetadataServiceUrl = url.URL{
+		Scheme: "http",
+		Host:   "[fe80::a9fe:a9fe]",
```

Comment:
I wonder if we might need to specify the interface name like it's mentioned on the docs 
https://docs.openstack.org/nova/latest/user/metadata.html#the-metadata-service
and how it's done elsewhere
https://github.com/canonical/cloud-init/blob/main/cloudinit/sources/DataSourceOpenStack.py#L76